src/pickers/datepicker.js

```diff
@@ -0,0 +1,1284 @@
+ */
+
+(function() {
+  'use strict';
+
+  /**
+   * Class constructor for date picker MDL component.
+   * Implements MDL component design pattern defined at:
+   * https://github.com/jasonmayes/mdl-component-design-pattern
+   *
+   * @constructor
+   * @param {HTMLElement} element The element that will be upgraded.
+   */
+  var MaterialDatePicker = function MaterialDatePicker(element) {
+    this.element_ = element;
+
+    // Initialize instance.
+    this.init();
+  };
+  window['MaterialDatePicker'] = MaterialDatePicker;
+
+  /**
+   * Global date picker settings
+   * Currently allowed formats: ['mm/dd/yyyy', 'dd.mm.yyyy', 'yyyy-mm-dd']
+   * @type {Object}
+   * @public
+   */
+  MaterialDatePicker.settings = {
+    /**
+     * Date format for input formatting
+     * @type {string}
+     */
+    format: 'yyyy-mm-dd',
+    weekDays: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
+    weekDaysShort: ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'],
+    weekDaysLetter: ['S', 'M', 'T', 'W', 'T', 'F', 'S'],
+    months: [
+      'January', 'February', 'March',
+      'April', 'May', 'June', 'July',
+      'August', 'September', 'October',
+      'November', 'December'
+    ],
+    monthsShort: [
+      'Jan', 'Feb', 'Mar', 'Apr', 'May',
+      'Jun', 'Jul', 'Aug', 'Sep', 'Oct',
+      'Nov', 'Dec'
+    ]
+  };
+
+  /**
+   * Instance based date picker settings.
+   * Overrides global date picker settings
+   *
+   * @type {Object}
+   * @public
+   */
+  MaterialDatePicker.prototype.settings = {};
+
+  /**
+   * Store constants in one place so they can be updated easily.
+   *
+   * @enum {string | number}
+   * @private
+   */
+  MaterialDatePicker.prototype.Constant_ = {
+    // None for now.
+  };
+
+  /**
+   * Store events in one place so they can be updated easily.
+   *
+   * @enum {string}
+   * @private
+   */
+  MaterialDatePicker.prototype.Event_ = {
+    // None for now.
+  };
+
+  /**
+   * Store strings for class names defined by this component that are used in
+   * JavaScript. This allows us to simply change it in one place should we
+   * decide to modify at a later date.
+   *
+   * @enum {string}
+   * @private
+   */
+  MaterialDatePicker.prototype.CssClasses_ = {
+    // General component classes
+    IS_UPGRADED: 'is-upgraded',
+    IS_VISIBLE: 'is-visible',
+    IS_ATTACHED: 'is-attached',
+    IS_INVALID: 'is-invalid',
+    IS_FOCUSED: 'is-focused',
+    IS_DIRTY: 'is-dirty',
+
+    // Appearance classes
+    INLINE: 'mdl-datepicker--inline',
+    FIXED: 'mdl-datepicker--fixed',
+    LANDSCAPE: 'mdl-datepicker--landscape',
+
+    // Datepicker related classes
+    BACKDROP: 'mdl-datepicker__backdrop',
+    WIDGET: 'mdl-datepicker__widget',
+    INPUT: 'mdl-datepicker__input',
+    NAVIGATION: 'mdl-datepicker__navigation',
+    CALENDAR: 'mdl-datepicker__calendar',
+    YEAR: 'mdl-datepicker__year',
+    YEAR_SELECTED: 'mdl-datepicker__year--selected',
```

Comment:
This looks like a status rather than an option, correct? We usually use `mdl-component--option` for options (or `mdl-component__part--option` for options on parts of the component) and `is-state` or `has-property` for transient states. The distinction being that options are something that users are supposed to set, whereas states are something that the component itself sets and unsets.

I'd suggest `is-year-selected` or `has-year-selected` in this case.